controllers/fenceagentsremediation_controller.go

```diff
@@ -68,69 +75,68 @@ type FenceAgentsRemediationReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.2/pkg/reconcile
 func (r *FenceAgentsRemediationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	r.Log.Info("started reconcile")
-	defer r.Log.Info("finished reconcile")
+	r.Log.Info("Begin FenceAgentsRemediation Reconcile")
+
+	// Fetch the FenceAgentsRemediationTemplate instance
+	emptyResult := ctrl.Result{}
+	key := client.ObjectKey{Namespace: req.Namespace, Name: fenceAgentsTemplateName}
+	farTemplate := &v1alpha1.FenceAgentsRemediationTemplate{}
+	if err := r.Client.Get(ctx, key, farTemplate); err != nil {
+		if apiErrors.IsNotFound(err) {
+			// FAR Template is not found, stop reconciling
+			r.Log.Info("FAR Template CR is not found - nothing to do", "CR Name", req.Name, "CR Namespace", req.Namespace)
+			return emptyResult, nil
+		}
+		r.Log.Error(err, "failed to get FAR Template CR")
+		return emptyResult, err
+	}
 
+	// Fetch the FenceAgentsRemediation instance
+	r.Log.Info("Fetch FAR CR")
 	far := &v1alpha1.FenceAgentsRemediation{}
-	if err := r.Get(ctx, req.NamespacedName, far); err != nil {
+	if err := r.Client.Get(ctx, req.NamespacedName, far); err != nil {
 		if apiErrors.IsNotFound(err) {
 			// FAR is deleted, stop reconciling
-			r.Log.Info("Fence Agents Remediation not found, nothing to do")
-			return ctrl.Result{}, nil
+			r.Log.Info("FAR CR is deleted - nothing to do", "CR Name", req.Name, "CR Namespace", req.Namespace)
+			return emptyResult, nil
 		}
-		r.Log.Error(err, "failed to get FAR")
-		return ctrl.Result{}, err
+		r.Log.Error(err, "failed to get FAR CR")
+		return emptyResult, err
 	}
-	key := client.ObjectKey{Namespace: req.Namespace, Name: fenceAgentsTemplateName}
-	farTemplate := &v1alpha1.FenceAgentsRemediationTemplate{}
-	if err := r.Get(ctx, key, farTemplate); err != nil {
-		r.Log.Error(err, "failed to get FAR template")
-		return ctrl.Result{}, err
+	// TODO: Validate FAR CR name to nodeName. Run isNodeNameValid
+	// Fetch the FAR's pod
+	r.Log.Info("Fetch FAR's pod")
+	pod, err := r.getFenceAgentsPod(req.Namespace)
+	if err != nil {
+		return emptyResult, err
 	}
 
-	pod, err := r.getFAPod(req.NamespacedName.Namespace)
+	// Build CLI executer for FAR's pod
+	r.Log.Info("Build CLI executer for FAR's pod")
+	ex, err := r.ExecutorHook(pod)
 	if err != nil {
-		return ctrl.Result{}, err
+		return emptyResult, err
 	}
 
-	ex, err := cli.NewExecuter(pod)
+	//TODO: Check that FA is excutable? run cli.IsExecuteable
+
+	r.Log.Info("Create and execute the fence agent", "Fence Agent", farTemplate.Spec.Agent)
```

Comment:
Nit: IMO this log is an overkill for each time reconcile is triggered.
(Keep in mind that we already have a log in the executor that has this info an more)